src/ui/ui.js

```diff
@@ -205,4 +207,74 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         }
     });
 
+    /**
+     * Updates the order of each block model according to the order of the blockUi
+     * elements in the story content editing area
+     *
+     * @param {Component} that - an instance of sjrk.storyTelling.ui.storyUi
+     * @param {Object} completionEvent - the event to be fired upon successful completion
+     */
+    sjrk.storyTelling.ui.updateBlockOrder = function (that, completionEvent) {
+        var blockUis = that.reorderer.container.children();
+        var loopCounter = 0;
+        var numberOfBlocks = blockUis.length;
+
+        // used to retrieve the class name for each block
+        // within the blockManager's managedViewComponentRegistry
+        var managedClassNamePattern = sjrk.storyTelling.ui.getManagedClassNamePattern(that.blockManager);
+
+        fluid.each(blockUis, function (blockUi) {
+            var managedClassName = fluid.find(blockUi.classList, function (value) {
```

Comment:
I haven't traced through all the details of this, but is the predicate here attempting to have the effect of https://api.jquery.com/hasClass/ ?

Reply:
What I attempted to do is to get a unique class name from the collection that matched a certain pattern, and then feed that class name into the `blockManager`'s component registry to get a reference to the `blockUi.editor` component that matches that class name, in order to reach its `block` subcomponent. I effectively wrote startsWith, though I think that's not supported in IE. Whether we're still supporting IE is something I'm unsure of.

I don't think `hasClass()` does what I want (I could be wrong), but if you know of any other approaches that fit the bill I would be glad to learn!

Reply:
Infusion is no longer explicitly supporting IE. We haven't removed code for it, but will no longer be testing in IE or actively writing code to support it. That means that any new release (pre-release or stable) cannot be relied on to support IE. So it may be time to revisit and/or define the browser support for Story Telling.

Reply:
@BlueSlug if I'm following the logic of the function, what you want to do is to take the DOM position of an element, find what Infusion component corresponds to that element, and then update the model values of that component with the position index. So maybe the first question is to @amb26 about whether there is an easy way to find a component from its container.

Another question is about the approach as a whole. That is you seem to only need this to update the disabled/enabled state of the up/down arrows for reordering. Here are some thoughts to get your started, none of them may be good though: 
- You could continue with the DOM based approach and on reorder have the components inspect where they are in the parents list of children or maybe if they have a previous or next sibling. In general modelizing state is better. 
- Should a reorderable element care about it's position? Should a single block UI really have to know that it is reorderable and where it is in the list? Perhaps this reordering behaviour could be the responsibility of a parent component that sort of decorates its children with the ability to be reordered. And there by managing the state of the individual reorderable elements.

Reply:
Code to maintain an index of viewComponent containers to their components can be adapted from the "viewMapper" in fluidViewDebugging.js here - https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/FluidViewDebugging.js#L576
You can simplify this code considerably since there is extra material here to account for the possibility that the view component lies in an iframe.
However, I agree with @jobara 's comment that responsibility for all list-based functionality should be centralised in a parent component.

Reply:
> Infusion is no longer explicitly supporting IE. We haven't removed code for it, but will no longer be testing in IE or actively writing code to support it. That means that any new release (pre-release or stable) cannot be relied on to support IE. So it may be time to revisit and/or define the browser support for Story Telling.

I see no reason to support IE in the Storytelling Tool at this point. We have so much to do, so anything we can do to simplify development seems like a good idea. Let's not spend time trying to support Internet Explorer.